Automattic/wp-calypso
# Reader: switch to Redux actions for site blocking

In https://github.com/Automattic/wp-calypso/pull/10649 we added site blocks to the Redux state tree.

This PR switches Reader components to use Redux actions and state for site blocking, rather than the old Flux store.

### To test

On a test account, follow some sites that aren't Discover.

In the post options menu, hit Block Site:

<img width="271" alt="screen shot 2017-01-31 at 17 53 25" src="https://cloud.githubusercontent.com/assets/17325/22477493/3716b526-e7de-11e6-8137-993038e23dee.png">

then in the resulting 'blocked post' card, make sure 'undo' works:

<img width="556" alt="screen shot 2017-01-31 at 17 54 09" src="https://cloud.githubusercontent.com/assets/17325/22477510/48a28fcc-e7de-11e6-8003-7999a97b0be5.png">

You should see an API request to `/me/block/sites/x/new` on block, and `/me/block/sites/x/delete` on unblock.

If you're an Automattician, you can also test the failure of blocks by altering the `WPCOM_JSON_API_Me_Block_Site_Endpoint` in your sandbox. Try sending a 500 error on block or unblock, or returning `success: false` in the response.
